perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -6058,6 +6087,43 @@ public void checkAttributeDependencies(PerunSession sess, RichAttribute richAttr
 	}
 
 	/**
+	 * Checks if the attributes represent empty values. If so, converts them into null.
+	 *
+	 * @param attributes attributes to be checked if are null
+	 */
+	public void convertEmptyAttrValueToNull(List<Attribute> attributes) {
+		attributes.forEach(this::convertEmptyAttrValueToNull);
+	}
+
+	/**
+	 * Checks if the attribute represents empty value. If so, converts it into null.
+	 *
+	 * @param attribute attribute to be checked if is null
+	 */
+	public void convertEmptyAttrValueToNull(Attribute attribute) {
+		if(attribute.getValue() == null) {
+			return;
+		}
+		if (attribute.getValue() instanceof String) {
+			if(((String)attribute.getValue()).matches("\\s*")) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof Boolean) {
+			if(attribute.getValue().equals(Boolean.FALSE)) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof ArrayList) {
+			if(((ArrayList)attribute.getValue()).isEmpty()) {
+				attribute.setValue(null);
+			}
+		} else if (attribute.getValue() instanceof HashMap) {
```

Comment:
Hi @zlamalp, I have just done these changes and after that some tests failed because they use an Integer attribute value type. Should I add another branch for Integer?